### PR TITLE
Teams

### PR DESCRIPTION
**Pull request by JPonte**
Team tabs and conversations.
The StartUI was also converted to Scala. This still requires some extra work to remove the java api and separate some of the logic from the views.
#### APK
[Download build #8931](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8931/artifact/build/artifact/wire-dev-PR894-8931.apk)
[Download build #8934](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8934/artifact/build/artifact/wire-dev-PR894-8934.apk)
[Download build #8935](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8935/artifact/build/artifact/wire-dev-PR894-8935.apk)
[Download build #8936](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8936/artifact/build/artifact/wire-dev-PR894-8936.apk)
[Download build #8937](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8937/artifact/build/artifact/wire-dev-PR894-8937.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/viewholders/AddressBookSectionHeaderViewHolder.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/**
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/viewholders/AddressBookSectionHeaderViewHolder.scala]: You know the drill...

File: app/src/main/scala/com/waz/zclient/viewholders/SectionExpanderViewHolder.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/viewholders/SectionExpanderViewHolder.scala]: ...and here....

File: app/src/main/scala/com/waz/zclient/views/SearchResultUserRowView.scala [hunk 3]
```diff
@@ -0,0 +1,97 @@
+/**
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/views/SearchResultUserRowView.scala]: ...

File: app/src/main/scala/com/waz/zclient/views/pickuser/ContactRowView.scala [hunk 4]
```diff
@@ -0,0 +1,197 @@
+/**
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/views/pickuser/ContactRowView.scala]: æøå

File: app/src/main/scala/com/waz/zclient/views/pickuser/ConversationRowView.scala [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/**
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/views/pickuser/ConversationRowView.scala]: ...

File: app/src/main/scala/com/waz/zclient/views/pickuser/SearchBoxView.scala [hunk 6]
```diff
@@ -0,0 +1,197 @@
+/**
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/views/pickuser/SearchBoxView.scala]: ...

File: app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+  var text = ""
+  var corners = UserCorners
+
+  val borderPaint = new Paint(Paint.ANTI_ALIAS_FLAG)
+  borderPaint.setColor(Color.TRANSPARENT)
+  borderPaint.setStyle(Paint.Style.STROKE)
+  borderPaint.setStrokeJoin(Paint.Join.ROUND)
+  borderPaint.setStrokeCap(Paint.Cap.ROUND)
+  borderPaint.setDither(true)
+  borderPaint.setPathEffect(new CornerPathEffect(10f))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala]: you can use `returning` here

File: app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+  val innerPaint = new Paint(Paint.ANTI_ALIAS_FLAG)
+  innerPaint.setColor(Color.TRANSPARENT)
+  innerPaint.setStyle(Paint.Style.FILL)
+  innerPaint.setStrokeJoin(Paint.Join.ROUND)
+  innerPaint.setStrokeCap(Paint.Cap.ROUND)
+  innerPaint.setDither(true)
+  innerPaint.setPathEffect(new CornerPathEffect(8f))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala]: `returning`

File: app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+  val textPaint = new Paint(Paint.ANTI_ALIAS_FLAG)
+  textPaint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.CLEAR))
+  textPaint.setTextAlign(Paint.Align.CENTER)
+  textPaint.setColor(Color.TRANSPARENT)
+  textPaint.setAntiAlias(true)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala]: `returning`

File: app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+  val bitmapPaint = new Paint(Paint.ANTI_ALIAS_FLAG)
+  bitmapPaint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/drawables/TeamIconDrawable.scala]: `returning`

File: app/src/main/scala/com/waz/zclient/fragments/ConversationListFragment.scala [hunk 11]
```diff
@@ -223,34 +207,73 @@ class NormalConversationFragment extends ConversationListFragment {
...
-  override def init(adapter: ConversationListAdapter, listActionsView: ListActionsView, topToolbar: ConversationListTopToolbar): Unit = {
-    adapter.currentMode ! ConversationListAdapter.Normal
 
-    hasArchive.on(Threading.Ui) { listActionsView.setArchiveEnabled }
+  override def init(view: View, adapter: ConversationListAdapter): Unit = {
+    val conversationListView = ViewUtils.getView(view, R.id.conversation_list_view).asInstanceOf[SwipeListView]
+    val listActionsView = ViewUtils.getView(view, R.id.lav__conversation_list_actions).asInstanceOf[ListActionsView]
+    val topToolbar = ViewUtils.getView(view, R.id.conversation_list_top_toolbar).asInstanceOf[NormalTopToolbar]
+    val noConvsTitle = ViewUtils.getView(view, R.id.conversation_list_empty_title).asInstanceOf[TypefaceTextView]
+    val noConvsSubtitle = ViewUtils.getView(view, R.id.conversation_list_empty_subtitle).asInstanceOf[TypefaceTextView]
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/fragments/ConversationListFragment.scala]: Is there no method in `FragmentHelper` that allows you to just do `getView[T](view, R.id....)`?

File: app/src/main/scala/com/waz/zclient/views/TeamTabButton.scala [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+class TeamTabButton(val context: Context, val attrs: AttributeSet, val defStyleAttr: Int) extends FrameLayout(context, attrs, defStyleAttr) with ViewHelper {
+  def this(context: Context, attrs: AttributeSet) = this(context, attrs, 0)
+  def this(context: Context) = this(context, null)
+
+  inflate(R.layout.view_team_tab)
+  setLayoutParams(new LayoutParams(context.getResources.getDimensionPixelSize(R.dimen.teams_tab_width), ViewGroup.LayoutParams.MATCH_PARENT))
+
+  val icon = ViewUtils.getView(this, R.id.team_icon).asInstanceOf[ImageView]
+  val name = ViewUtils.getView(this, R.id.team_name).asInstanceOf[TypefaceTextView]
+  val unreadIndicator = ViewUtils.getView(this, R.id.unread_indicator).asInstanceOf[CircleView]
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/TeamTabButton.scala]: `findById[Type](id)`

File: app/src/main/scala/com/waz/zclient/views/conversationlist/ConversationListTopToolbar.scala [hunk 13]
```diff
@@ -20,36 +20,103 @@ package com.waz.zclient.views.conversationlist
...
+
+  private implicit val logTag = ZLog.logTagFor[ConversationListTopToolbar]
 
   inflate(R.layout.view_conv_list_top)
 
   val bottomBorder = ViewUtils.getView(this, R.id.conversation_list__border).asInstanceOf[View]
   val glyphButton = ViewUtils.getView(this, R.id.conversation_list_settings).asInstanceOf[GlyphTextView]
   val title = ViewUtils.getView(this, R.id.conversation_list_title).asInstanceOf[TypefaceTextView]
   val settingsIndicator = ViewUtils.getView(this, R.id.conversation_list_settings_indicator).asInstanceOf[CircleView]
+  val tabsContainer = ViewUtils.getView(this, R.id.team_tabs_container).asInstanceOf[TeamTabsView]
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/conversationlist/ConversationListTopToolbar.scala]: `findById[T](id)`

File: app/src/main/scala/com/waz/zclient/views/pickuser/ContactRowView.scala [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+//TODO: remove the uiObservable for contact?
+class ContactRowView(val context: Context, val attrs: AttributeSet, val defStyleAttr: Int) extends FrameLayout(context, attrs, defStyleAttr) with UserRowView with ViewHelper with Injectable {
+  def this(context: Context, attrs: AttributeSet) = this(context, attrs, 0)
+  def this(context: Context) = this(context, null)
+
+  inflate(R.layout.list_row_contactlist_user, this, addToParent = true)
+
+  private val chathead: ChatheadView = ViewUtils.getView(this, R.id.cv__contactlist__user__chathead)
+  private val contactListItemTextView: ContactListItemTextView = ViewUtils.getView(this, R.id.clitv__contactlist__user__text_view)
+  private val contactInviteButton: ZetaButton = ViewUtils.getView(this, R.id.zb__contactlist__user_selected_button)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/pickuser/ContactRowView.scala]: `findById[T](id)`

File: app/src/main/scala/com/waz/zclient/views/pickuser/SearchBoxView.scala [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+
+  def this(context: Context) {
+    this(context, null)
+  }
+
+  private def init(context: Context): Unit = {
+    LayoutInflater.from(context).inflate(R.layout.search_box_view, this, true)
+    clearButton = ViewUtils.getView(this, R.id.gtv_pickuser__clearbutton)
+    inputEditText = ViewUtils.getView(this, R.id.puet_pickuser__searchbox)
+    colorBottomBorder = ViewUtils.getView(this, R.id.v_people_picker__input__color_bottom_border)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/pickuser/SearchBoxView.scala]: `findById[T](id)`

**Pull request merged**